misc/Makefile

```diff
@@ -0,0 +1,131 @@
+uname_M := $(shell uname -m 2>/dev/null || echo not)
+
+ARCH ?= $(shell echo $(uname_M) | sed -e s/i.86/i386/ -e s/arm.*/arm/ )
+ifeq ($(ARCH),x86_64)
+  ifneq ($(findstring m32,$(CFLAGS)),)
+    override ARCH := i386
+  endif
+endif
+
+ifndef ($(srcdir))
+	srcdir = $(CURDIR)/..
+endif
+# set objdir to $(O) by default (if any)
+ifndef ($(objdir))
+  objdir = $(CURDIR)/..
+endif
+
+ifneq ($(wildcard $(objdir)/.config),)
+  include $(objdir)/.config
+endif
+
+RM = rm -f
+
+COMMON_CFLAGS := -D_GNU_SOURCE $(CFLAGS) $(CPPFLAGS)
+COMMON_CFLAGS += -iquote $(srcdir) -iquote $(objdir) -iquote $(srcdir)/arch/$(ARCH)
+COMMON_LDFLAGS := -lrt -ldl -pthread $(LDFLAGS)
+ifneq ($(elfdir),)
+  COMMON_CFLAGS  += -I$(elfdir)/include
+  COMMON_LDFLAGS += -L$(elfdir)/lib
+endif
+
+COMMON_CFLAGS += -W -Wall -Wno-unused-parameter -Wno-missing-field-initializers
```

Comment:
Do these need to be repeated?  I think you can export them in the main Makefile and just use it here.